### ISSUE TITLE
"upgrade-all-service-instances" exits 0 when upgrading an instance fails

### ISSUE THREAD

**Issue by degaurab** [pivotal-cf/on-demand-service-broker-release#3]
We would expect the errand to exit non-zero if any instance failed to upgrade, but this was not the case:
```
Using deployment 'dedicated-mysql-broker'

Task 3243435

17:07:19 | Preparing deployment: Preparing deployment
17:07:22 | Preparing package compilation: Finding packages to compile (00:00:00)
17:07:22 | Preparing deployment: Preparing deployment (00:00:03)
17:07:22 | Running errand: upgrade-all-service-instances/0 (00:40:12)
17:47:34 | Fetching logs for upgrade-all-service-instances/78fd0c4a-4446-4aa7-983c-a7380888e86e (0): Finding and packing log files (00:00:01)

Started  Tue May 30 17:07:19 UTC 2017
Finished Tue May 30 17:47:35 UTC 2017
Duration 00:40:16

Task 3243435 done

Exit Code  Stdout                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Stderr  
0          [upgrade-all-service-instances] 2017/05/30 17:07:22.425417 STARTING UPGRADES                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 -       
           [upgrade-all-service-instances] 2017/05/30 17:07:24.007178 Service Instances: zzz-f780-45  85d9ea74-xxxxxxxx
....... b3a30744-52cd-4bf2-b417-xxxx          
           [upgrade-all-service-instances] 2017/05/30 17:07:24.007217 Total Service Instances found in Cloud Foundry: 22                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
           [upgrade-all-service-instances] 2017/05/30 17:07:24.007227 Service instance: xxxxx-f780-45ff-9c89-xxxxx, upgrade attempt starting (1 of 22)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
           [upgrade-all-service-instances] 2017/05/30 17:07:24.325367 Result: accepted upgrade                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
           [upgrade-all-service-instances] 2017/05/30 17:07:24.325396 Waiting for upgrade to complete for xxxxx-f780-45ff-9c89-xxxx: bosh task id 3243436                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
           [upgrade-all-service-instances] 2017/05/30 17:09:24.344353 Result: Service Instance xxxx-f780-45ff-9c89-xxxx upgrade success

...
...
...

           [upgrade-all-service-instances] 2017/05/30 17:44:33.679189 Service instance: xx-eaea-4564-a528-xxxx, upgrade attempt starting (19 of 22)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
           [upgrade-all-service-instances] 2017/05/30 17:44:34.165106 Result: accepted upgrade                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
           [upgrade-all-service-instances] 2017/05/30 17:44:34.165122 Waiting for upgrade to complete for xx-eaea-4564-a528-xxxx: bosh task id 3244088                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
           [upgrade-all-service-instances] 2017/05/30 17:47:34.193725 Result: Service Instance xx-eaea-4564-a528-xxxx upgrade failure                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
           [upgrade-all-service-instances] 2017/05/30 17:47:34.193739 Upgrade failed for service instance xx-eaea-4564-a528-xxxx: bosh task id 3244088: Failed for bosh task: 3244088                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             


1 errands

Succeeded
```

CF Dedicated MySQL team
/cc @mfine30 & @youreddy

**Comment by avade** [comment]
Thanks for raising this issue - which ODB version are you using?

**Comment by youreddy** [comment]
@avade, currently using `0.15.3` on prod (where this pair saw the issue)

**Comment by avade** [comment]
Resolved in 0.16.1
